File: Frontend/lib/models/invigilation.dart
import 'enums.dart';

class Invigilation {
  final String subject;
  final String room;
  final String time;
  String? faculty;
  InvigilationStatus status;

  Invigilation({
    required this.subject,
    required this.room,
    required this.time,
    this.faculty,
    this.status = InvigilationStatus.unassigned,
  });
}

File: Backend/app/models/invigilation.py
from pydantic import BaseModel
from typing import Optional
from datetime import date, time
from enum import Enum

class InvigilationStatus(str, Enum):
    unassigned = "unassigned"
    assigned = "assigned"

class Invigilation(BaseModel):
    id: Optional[str] = None
    exam_name: str
    subject: str
    room: str
    date: date
    start_time: time
    end_time: time
    faculty_id: Optional[str] = None
    status: InvigilationStatus = InvigilationStatus.unassigned